larpandoracontent/LArMonitoring/VertexAssessmentAlgorithm.cc

```diff
@@ -0,0 +1,209 @@
+                                const CartesianVector trueW(trueVertex.GetX(), 0.f, static_cast<float>(transform->YZtoW(trueVertex.GetY(), trueVertex.GetZ())));
+                                PANDORA_MONITORING_API(AddMarkerToVisualization(this->GetPandora(), &trueU, "true_vtx_u", ORANGE, 2));
+                                PANDORA_MONITORING_API(AddMarkerToVisualization(this->GetPandora(), &trueV, "true_vtx_v", ORANGE, 2));
+                                PANDORA_MONITORING_API(AddMarkerToVisualization(this->GetPandora(), &trueW, "true_vtx_w", ORANGE, 2));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    catch (const StatusCodeException &)
+    {   // No MC available
+    }
+
+    const VertexList *pVertexList{nullptr};
+    if (STATUS_CODE_SUCCESS == PandoraContentApi::GetList(*this, m_vertexListName, pVertexList))
+    {
+        if (pVertexList && !pVertexList->empty())
+        {
+            for (const Vertex *pVertex : *pVertexList)
+            {
+                const CartesianVector recoVertex(pVertex->GetPosition());
+                if (pVertex->GetVertexLabel() == VERTEX_INTERACTION)
+                {
+                    if (m_visualize)
+                    {
+                        const CartesianVector recoU(recoVertex.GetX(), 0.f, static_cast<float>(transform->YZtoU(recoVertex.GetY(), recoVertex.GetZ())));
+                        const CartesianVector recoV(recoVertex.GetX(), 0.f, static_cast<float>(transform->YZtoV(recoVertex.GetY(), recoVertex.GetZ())));
+                        const CartesianVector recoW(recoVertex.GetX(), 0.f, static_cast<float>(transform->YZtoW(recoVertex.GetY(), recoVertex.GetZ())));
+                        PANDORA_MONITORING_API(AddMarkerToVisualization(this->GetPandora(), &recoU, "reco_vtx_u", RED, 2));
+                        PANDORA_MONITORING_API(AddMarkerToVisualization(this->GetPandora(), &recoV, "reco_vtx_v", GREEN, 2));
+                        PANDORA_MONITORING_API(AddMarkerToVisualization(this->GetPandora(), &recoW, "reco_vtx_w", BLUE, 2));
+                    }
+                    if (truthAvailable && isReconstructable)
+                    {
+                        const float dr{std::sqrt(recoVertex.GetDistanceSquared(trueVertex))};
+                        std::cout << "dr: " << dr << std::endl;
+                        PANDORA_MONITORING_API(SetTreeVariable(this->GetPandora(), "vertex", "dr", dr));
+                        PANDORA_MONITORING_API(FillTree(this->GetPandora(), "vertex"));
+                    }
+                }
+            }
+        }
+        else
+        {
+            if (truthAvailable && isReconstructable)
+            {   // Failed to find a vertex, apply large distance penalty
+                const float dr{5000.f};
```

Comment:
why {}

Reply:
I generally prefer {} initialisation to protect against narrowing conversions